crates/bevy_input/src/state.rs

```diff
@@ -0,0 +1,32 @@
+/// The current state of an element.
+///
+/// ## Usage
+///
+/// It is used to define the state of the [`KeyboardInput`](crate::keyboard::KeyboardInput) and
```

Comment:
Shouldn't this also be used for `GamepadButton`?

Reply:
`KeyboardInput` and `MouseButtonInput` are events, whereas `GamepadButton` is a resource. If we wanted to add it to the `GamepadEvent` we would probably have to add it to the `GamepadEventType::ButtonChanged` as `(GamepadButtonType, ButtonState)`, but I'm not sure how easy it would be to do that, since we also have things like triggers inside of `GamepadButtonType`.

This might be something to consider inside of your PR #3419.

Reply:
Err right, `GamepadButtonType`. Alright, I'll take a look again once I'm working on #3419 again.